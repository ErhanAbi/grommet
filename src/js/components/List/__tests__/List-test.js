import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { cleanup, render, fireEvent } from '@testing-library/react';

import { Grommet } from '../../Grommet';
import { List } from '..';

describe('List', () => {
  afterEach(cleanup);

  test('empty', () => {
    const component = renderer.create(
      <Grommet>
        <List />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('strings', () => {
    const component = renderer.create(
      <Grommet>
        <List data={['one', 'two']} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('objects', () => {
    const component = renderer.create(
      <Grommet>
        <List data={[{ a: 'one', b: 1 }, { a: 'two', b: 2 }]} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('click', () => {
    const onClickItem = jest.fn();
    const { container, getByText } = render(
      <Grommet>
        <List
          data={[{ a: 'alpha' }, { a: 'beta' }]}
          onClickItem={onClickItem}
        />
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
    fireEvent.click(getByText('beta'));
    expect(onClickItem).toBeCalledWith(
      expect.objectContaining({ item: { a: 'beta' } }),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
