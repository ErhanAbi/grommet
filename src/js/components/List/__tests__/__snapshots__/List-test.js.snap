// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List click 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 0px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-top: solid 1px rgba(0,0,0,0.33);
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c1 {
  list-style: none;
  margin: 0;
  padding: 0;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    border-top: solid 1px rgba(0,0,0,0.33);
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}

<div
  class="c0"
>
  <ol
    class="c1"
    tabindex="0"
  >
    <li
      class="c2"
    >
      <button
        class="c3"
        type="button"
      >
        <div
          class="c4"
        >
          alpha
        </div>
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        type="button"
      >
        <div
          class="c5"
        >
          beta
        </div>
      </button>
    </li>
  </ol>
</div>
`;

exports[`List click 2`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 eUfLDp"
>
  <ol
    class="List__StyledList-sc-130gdqg-0 ilqdOl"
    tabindex="0"
  >
    <li
      class="StyledBox-sc-13pk1d4-0 hqGUhX"
    >
      <button
        class="List__StyledButton-sc-130gdqg-1 aAhAP StyledButton-sc-323bzc-0 jfbKsv"
        type="button"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 krTYKC"
        >
          alpha
        </div>
      </button>
    </li>
    <li
      class="StyledBox-sc-13pk1d4-0 hqGUhX"
    >
      <button
        class="List__StyledButton-sc-130gdqg-1 aAhAP StyledButton-sc-323bzc-0 jfbKsv"
        type="button"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 gSGywF"
        >
          beta
        </div>
      </button>
    </li>
  </ol>
</div>
`;

exports[`List empty 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  list-style: none;
  margin: 0;
  padding: 0;
}

<div
  className="c0"
>
  <ol
    className="c1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`List objects 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-top: solid 1px rgba(0,0,0,0.33);
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c1 {
  list-style: none;
  margin: 0;
  padding: 0;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    border-top: solid 1px rgba(0,0,0,0.33);
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}

<div
  className="c0"
>
  <ol
    className="c1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <li
      className="c2"
    >
      one
    </li>
    <li
      className="c3"
    >
      two
    </li>
  </ol>
</div>
`;

exports[`List strings 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-top: solid 1px rgba(0,0,0,0.33);
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c1 {
  list-style: none;
  margin: 0;
  padding: 0;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    border-top: solid 1px rgba(0,0,0,0.33);
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}

<div
  className="c0"
>
  <ol
    className="c1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  >
    <li
      className="c2"
    >
      one
    </li>
    <li
      className="c3"
    >
      two
    </li>
  </ol>
</div>
`;
